src/Sylius/Behat/Context/Ui/Admin/ManagingShippingMethodsContext.php

```diff
@@ -12,87 +12,420 @@
+        CreatePageInterface $createPage,
+        UpdatePageInterface $updatePage,
+        CurrentPageResolverInterface $currentPageResolver,
+        NotificationCheckerInterface $notificationChecker
     ) {
         $this->sharedStorage = $sharedStorage;
-        $this->shippingMethodShowPage = $shippingMethodDetailsPage;
-        $this->shippingMethodIndexPage = $shippingMethodIndexPage;
+        $this->indexPage = $indexPage;
+        $this->createPage = $createPage;
+        $this->updatePage = $updatePage;
+        $this->currentPageResolver = $currentPageResolver;
+        $this->notificationChecker = $notificationChecker;
     }
 
     /**
-     * @When /^I try to delete ("[^"]+" shipping method)$/
+     * @Given I want to create a new shipping method
      */
-    public function iTryToDeleteShippingMethod(ShippingMethodInterface $shippingMethod)
+    public function iWantToCreateANewShippingMethod()
     {
-        $this->sharedStorage->set('shipping_method', $shippingMethod);
+        $this->createPage->open();
+    }
 
-        $this->shippingMethodShowPage->open(['id' => $shippingMethod->getId()]);
-        $this->shippingMethodShowPage->pressDelete();
+    /**
+     * @When I specify its code as :code
+     * @When I do not specify its code
+     */
+    public function iSpecifyItsCodeAs($code = null)
+    {
+        $this->createPage->specifyCode($code);
     }
 
     /**
-     * @Then I should be notified that it is in use
+     * @When I name it :name in :language
+     * @When I rename it to :name in :language
+     */
+    public function iNameItIn($name, $language)
+    {
+        $this->createPage->nameIt($name, $language);
+    }
+
+    /**
+     * @When I define it for the :zoneName zone
+     */
+    public function iDefineItForTheZone($zoneName)
+    {
+        $this->createPage->chooseZone($zoneName);
+    }
+
+    /**
+     * @When I specify its amount as :amount
+     */
+    public function iSpecifyItsAmountAs($amount)
+    {
+        $this->createPage->specifyAmount($amount);
+    }
+
+    /**
+     * @When I add it
+     * @When I try to add it
+     */
+    public function iAddIt()
+    {
+        $this->createPage->create();
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully created
+     */
+    public function iShouldBeNotifiedItHasBeenSuccessfulCreation()
+    {
+        $this->notificationChecker->checkCreationNotification(self::RESOURCE_NAME);
+    }
+
+    /**
+     * @When I choose :calculatorName calculator
+     * @When I do not specify amount for :calculatorName calculator
+     */
+    public function iChooseCalculator($calculatorName)
+    {
+        $this->createPage->chooseCalculator($calculatorName);
+    }
+
+    /**
+     * @Then the shipment method :shipmentMethod should appear in the registry
+     * @Then the shipment method :shipmentMethod should be in the registry
+     * @Given /^(this shipping method) should still be in the registry$/
```

Comment:
I think it's incorrect, as it will return ``ShippingMethod`` object instead of it's name.